osu.Game.Tests/Gameplay/TestSceneDrainingHealthProcessor.cs

```diff
@@ -160,6 +160,43 @@ public void TestHealthRemovedOnRevert()
             assertHealthNotEqualTo(1);
         }
 
+        [Test]
+        public void TestFailConditions()
+        {
+            var beatmap = createBeatmap(0, 1000);
+            createProcessor(beatmap);
+
+            AddStep("setup fail conditions", () => processor.FailConditions += ((_, result) => result.Type == HitResult.Miss));
+
+            AddStep("apply perfect hit result", () => processor.ApplyResult(new JudgementResult(beatmap.HitObjects[0], new Judgement()) { Type = HitResult.Perfect }));
+            AddAssert("not failed", () => !processor.HasFailed);
+            AddStep("apply miss hit result", () => processor.ApplyResult(new JudgementResult(beatmap.HitObjects[0], new Judgement()) { Type = HitResult.Miss }));
+            AddAssert("failed", () => processor.HasFailed);
+        }
+
+        [Test]
+        public void TestMultipleFailConditions([Values] bool applyFirstCondition)
```

Comment:
I would have made the `HitResult` an argument of this test rather than a `bool`, as in

```csharp
[TestCase(HitResult.Miss)]
[TestCase(HitResult.Meh)]
public void TestMultipleFailConditions(HitResult result)
```

Generally dislike generic bool method arguments and recommend avoiding wherever possible and reasonable, almost any other type is preferable to passing a `bool`.

Reply:
Fair, also wasn't sure about this.